File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        if(s.size() == 0) 
            return s;
        stack<string> st;
        string ans;
        for(int i=0; i<s.size(); i++) 
        {
            string word;
            if(s[i] == ' ') continue; 
            
            while(i < s.size() && s[i] != ' ' ) { 
                word += s[i]; 
                i++;
            }
            
            st.push(word); 
        }
        while(!st.empty()) {
            ans += st.top(); 
            st.pop();
            if(!st.empty()) 
            {
                ans += ' ';
            }
        }
        return ans;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        if(s.length() == 0){
            return s;
        }
        Stack<String> st = new Stack<>();
        String ans ="";
        for(int i = 0; i<s.length(); i++){
            
            String word = "";
            if(s.charAt(i) == ' ') continue;
            
            while(i<s.length() && s.charAt(i) != ' '){
                word += s.charAt(i);
                i++;
            }
            st.push(word);
        }
        
        while(!st.isEmpty()){
            ans += st.pop();
            if(!st.isEmpty()){
                ans += ' ';
            }
        }
        return ans;
    }
}